src/GL/Array.hs

```diff
@@ -175,34 +171,14 @@ load is = do
 
 
 bindArray :: (Has Check sig m, Has (Lift IO) sig m) => Array (v I) -> ArrayC v m a -> m a
-bindArray array (ArrayC m) = do
+bindArray array m = do
   bind (Just array)
-  a <- runReader array m
+  a <- runReader array (runLabelled m)
   a <$ bind @(Array _) Nothing
 
-class Monad m => HasArray v m | m -> v where
-  askArray :: m (Array (v I))
+type HasArray v sig m = HasLabelled Array (Reader (Array (v I))) sig m
 
+askArray :: HasLabelled Array (Reader (Array (v I))) sig m => m (Array (v I))
+askArray = runUnderLabel @_ @Array ask
```

Comment:
All you have to do to label effects is use `runUnderLabel` to lift them & apply the desired label.

The first `@_` is to infer the kind of the label; I would like to make this unnecessary and only require you to specify the label parameter, but I’m not sure that’s possible with `PolyKinds`, and poly-kinded labels are pretty valuable.